backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java

```diff
@@ -259,7 +265,54 @@ private PrepareImageVDSCommandParameters getPrepareParameters(Guid vdsId) {
                 getDiskImage().getImageId(), true);
     }
 
+    private boolean validateBitmap(DiskImage image, Guid checkpointId) {
+        VdsmImageLocationInfo locationInfo = new VdsmImageLocationInfo(
+                    image.getStorageIds().get(0),
+                    image.getId(),
+                    image.getImageId(),
+                    null);
+
+        VolumeBitmapCommandParameters parameters =
+        new VolumeBitmapCommandParameters(
+                getStoragePoolId(),
+                locationInfo,
+                null);
+        parameters.setVdsId(getVdsId());
+        parameters.setEndProcedure(ActionParametersBase.EndProcedure.COMMAND_MANAGED);
+        parameters.setParentCommand(getActionType());
+        parameters.setParentParameters(getParameters());
+
+        ActionReturnValue returnValue = runInternalActionWithTasksContext(ActionType.ListVolumeBitmaps, parameters);
+        boolean valid = false;
+        if (returnValue.getSucceeded()) {
+            UUIDListReturn bitmaps = returnValue.getActionReturnValue();
+            valid = Arrays.stream(bitmaps.getUUIDList()).anyMatch(checkpointId.toString()::equals);
+        }
+        /* ListVolumeBitmaps failed or bitmap did not exist on disk -> Remove checkpoints */
+        if (!valid) {
+            log.error("Checkpoint '{}' does not exist on disk '{}'. Removing all checkpoints.",
+                    checkpointId,
+                    image.getId(),
+                    checkpointId);
+            /* Some checkpoint corruption, remove checkpoints */
+            DeleteAllVmCheckpointsParameters deleteAllVmCheckpointsParameters =
+            new DeleteAllVmCheckpointsParameters(getVmId(), List.of(image));
+            deleteAllVmCheckpointsParameters.setParentCommand(getActionType());
+            deleteAllVmCheckpointsParameters.setParentParameters(getParameters());
+            deleteAllVmCheckpointsParameters.setEndProcedure(EndProcedure.COMMAND_MANAGED);
+            deleteAllVmCheckpointsParameters.setForce(true);
+
+            runInternalAction(ActionType.DeleteAllVmCheckpoints, deleteAllVmCheckpointsParameters);
+            return false;
+        }
+        return true;
+    }
+
     private Guid getBitmap() {
+        if (!validateBitmap(getDiskImage(), getBackup().getFromCheckpointId())) {
```

Comment:
can we put it elsewhere, it feels out of place for a getter

Reply:
The best spot would be in the validate() function then? Or early in the executeCommand()